src/coreimage.cs

```diff
@@ -4729,70 +4733,76 @@ interface CIAttributedTextImageGenerator {
 	[TV (11,0)]
 	[BaseType (typeof (CIFilter))]
 	interface CIBarcodeGenerator {
-		// TODO: Needs review
-		//[CoreImageProperty ("inputBarcodeDescriptor")]
-		//CIBarcodeDescriptor BarcodeDescriptor { get; set; }
+		[CoreImageFilterProperty ("inputBarcodeDescriptor")]
+		CIBarcodeDescriptor BarcodeDescriptor { get; set; }
 	}
 
 	[CoreImageFilter]
 	[iOS (11,0)]
 	[Mac (10,13)]
 	[TV (11,0)]
-	[BaseType (typeof (CIFilter))]
+	[BaseType (typeof (CIFilter))] // Maybe 'CIScaleTransform' (shared 'Scale' and 'AspectRatio' property)?
 	interface CIBicubicScaleTransform {
-		// TODO: Needs review
-		//[CoreImageProperty ("inputB")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float B { get; set; }
-		//[CoreImageProperty ("inputScale")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float Scale { get; set; }
-		//[CoreImageProperty ("inputC")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float C { get; set; }
-		//[CoreImageProperty ("inputAspectRatio")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float AspectRatio { get; set; }
+		[CoreImageFilterProperty ("inputB")]
+		float B { get; set; }
+
+		[CoreImageFilterProperty ("inputC")]
+		float C { get; set; }
+
+		[CoreImageFilterProperty ("inputScale")]
+		float Scale { get; set; }
+
+		[CoreImageFilterProperty ("inputAspectRatio")]
+		float AspectRatio { get; set; }
 	}
 
 	[CoreImageFilter]
+	[Abstract]
 	[iOS (11,0)]
 	[Mac (10,13)]
 	[TV (11,0)]
 	[BaseType (typeof (CIFilter))]
+	interface CIBlur { // `CIMotionBlur.SuperClass == CILinearBlur` but it's marked as `[NoiOS]` and `[NoMac]` so we have ours
```

Comment:
same, does `CIBlur` exists ?

Reply:
I do not see `CILinearBlur` anywhere in this PR or the existing sources

Reply:
See https://github.com/xamarin/xamarin-macios/pull/2974#issuecomment-343296118, `CIBlur` doesn't exist but `CILinearBlur` does but it's marked as `[NoiOS]` and `[NoMac]` so I'm assuming it's not expected on iOS and Mac. That confuses me.

Reply:
it's been a while... and I remember mistakes in the values returned. However it could mean it's an _abstract_ type (so not _usable_, instead of not available, from iOS or Mac).